lessons/python/inttopics/inttopics.ipynb

```diff
@@ -0,0 +1,672 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "help(scores.append) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looks like what we want, but let's check on `insert` just to be sure -"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "help(scores.insert) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# append it is!\n",
+    "\n",
+    "scores.append(89)\n",
+    "\n",
+    "print(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It's possible to define a custom set of methods and attributes in order to create new **classes** of objects. We won't go into much depth about these here, though. (See: Lina's lesson on classes in Python)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## List Comprehensions\n",
+    "\n",
+    "Let's say we want to get the original scores, out of 40. Is there a built-in method we could use in order to do the same thing to every item in a list?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "dir(mylist)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Apparently not. Perhaps a for loop is the way to go? Let's test that out."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "for i in range(len(scores)):\n",
+    "    scores[i] = scores[i] * 0.4\n",
+    "    \n",
+    "print(scores)     "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Whoa, that's not right!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "print(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looks like Python imported the data as a list of strings, save for that one 89% we appended.\n",
+    "\n",
+    "We need to find a way to change those strings to integers instead, and then multiply them by 0.4."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# this is one way?\n",
+    "for num in scores:\n",
+    "    if type(num) is str:\n",
+    "        num = int(num)\n",
+    "        num = num * 0.4\n",
+    "    else:\n",
+    "        pass"
```

Comment:
I wanted to make a point just below this about how list comprehensions do not alter an object inplace like `append` does, but I suppose this is a bit of a clumsy way of leading to it. Should I just get rid of the else clause entirely?